confmap/provider/envprovider/provider.go

```diff
@@ -27,6 +27,14 @@ type provider struct {
 //
 // This Provider supports "env" scheme, and can be called with a selector:
 // `env:NAME_OF_ENVIRONMENT_VARIABLE`
+//
+// A default value for unset variable can be provided after :- suffix, for example:
+// `env:NAME_OF_ENVIRONMENT_VARIABLE:-default_value`
+//
+// An error message for unset variable can be provided after :? suffix, for example:
+// `env:NAME_OF_ENVIRONMENT_VARIABLE:?error_message`
```

Comment:
I'm not aware of a definition for setting an error message, is there an open issue somewhere?

Reply:
I think we should still have it. I don't understand the approach "let's be like Shell but just different enough so that common things don't work".

Reply:
not disagreeing, this was me asking if this is already a discussion about this somewhere I haven't found yet

Reply:
> I don't understand the approach "let's be like Shell but just different enough so that common things don't work".

This doesn't mean that we need to add full support for Bash/POSIX-like syntax today. Your PR does not add full support for [all the features that Bash supports](https://tiswww.case.edu/php/chet/bash/bashref.html#Shell-Expansions) and that's okay. As I said above, I am not opposed to adding this, but since alignment with the Configuration WG is an explicit goal we should bring it up on their repo first.